packages/components/src/inputs/src/InputGroup.module.css

```diff
@@ -0,0 +1,124 @@
+    --hop-InputGroup-input-sm-font-weight: var(--hop-body-sm-font-weight);
+    --hop-InputGroup-input-sm-line-height: var(--hop-body-sm-line-height);
+    --hop-InputGroup-input-sm-block-size: 1.5rem;
+    --hop-InputGroup-input-md-font-family: var(--hop-body-md-font-family);
+    --hop-InputGroup-input-md-font-size: var(--hop-body-md-font-size);
+    --hop-InputGroup-input-md-font-weight: var(--hop-body-md-font-weight);
+    --hop-InputGroup-input-md-line-height: var(--hop-body-md-line-height);
+    --hop-InputGroup-input-md-block-size: 2rem;
+    --hop-InputGroup-background-disabled: var(--hop-neutral-surface-disabled);
+    --hop-InputGroup-border-color-disabled: var(--hop-neutral-border-disabled);
+    --hop-InputGroup-placeholder-font-family: inherit;
+    --hop-InputGroup-placeholder-font-size: inherit;
+    --hop-InputGroup-placeholder-font-weight: inherit;
+    --hop-InputGroup-placeholder-line-height: inherit;
+    --hop-InputGroup-placeholder-color: var(--hop-neutral-text-weakest);
+    --hop-InputGroup-placeholder-color-disabled: var(--hop-neutral-text-disabled);
+
+    /* Internal Variables */
+    --inline-size: fit-content;
+    --border: var(--hop-InputGroup-border-width) solid var(--hop-InputGroup-border-color);
+    --background: var(--hop-InputGroup-background);
+    --block-size: var(--hop-InputGroup-md-block-size);
+
+    position: relative;
+
+    display: inline-flex;
+    gap: var(--hop-InputGroup-gap);
+    align-items: center;
+
+    box-sizing: border-box;
+    inline-size: var(--inline-size);
+    block-size: var(--block-size);
+    padding-inline: var(--hop-InputGroup-padding-inline);
+
+    background: var(--background);
+    border: var(--border);
+    border-radius: var(--hop-InputGroup-border-radius)
+}
+
+.hop-InputGroup--fluid {
+    --inline-size: 100%
+}
+
+.hop-InputGroup--fluid .hop-InputGroup__input {
+    inline-size: 100%;
+}
+
+.hop-InputGroup[data-hovered] {
+    --background: var(--hop-InputGroup-background-hover);
+    --border: var(--hop-InputGroup-border-width) solid var(--hop-InputGroup-border-color-hover)
+}
+
+.hop-InputGroup[data-focus-within]  {
+    --border: var(--hop-InputGroup-border-width) solid var(--hop-InputGroup-border-color-focus)
+}
+
+.hop-InputGroup[data-invalid] {
+    --border: var(--hop-InputGroup-border-width) solid var(--hop-InputGroup-border-color-invalid)
+}
+
+.hop-InputGroup--sm {
+    --block-size: var(--hop-InputGroup-sm-block-size);
+}
+
+.hop-InputGroup__input {
+    flex-grow: 1;
+
+    block-size: var(--hop-InputGroup-input-md-block-size);
+    padding: 0;
+
+    font-family: var(--hop-InputGroup-input-md-font-family);
+    font-size: var(--hop-InputGroup-input-md-font-size);
+    font-weight: var(--hop-InputGroup-input-md-font-weight);
+    line-height: var(--hop-InputGroup-input-md-line-height);
+    color: var(--hop-InputGroup-input-color);
+
+    background: transparent;
+    border: none;
+    outline: none
+}
+
+.hop-InputGroup--sm .hop-InputGroup__input {
```

Comment:
you shouldn't need to use .hop-InputGroup--sm before .hop-InputGroup__input.
All this properties can be internal variables prefixed with --input.